src/tsd/HttpQuery.java

```diff
@@ -64,7 +57,7 @@
  * It makes it easier to provide a few utility methods to respond to the
  * requests.
  */
-final class HttpQuery {
+final class HttpQuery extends AbstractHttpQuery {
```

Comment:
I moved a bunch more methods but the send-related ones are a little harder to tease apart due to their reliance on getting the MIME type from the serializer.  I can try to abstract out the determination of MIME type and push that up into the base class, but I worry that would push too much modeling up...